src/overlays/ovl_i5/fox_ti.c

```diff
@@ -1,138 +1,5375 @@
+            D_i5_801BBEF0[3]--;
+        }
+    }
+    if ((D_i5_801BBEF0[5] != 0) &&
+        (Math_SmoothStepToAngle(&D_i5_801BBEF4[8], D_i5_801BBEF4[9], 1.0f, 8.0f, 0.01f) == 0.0f)) {
+        if (D_i5_801BBEF4[9] == 48.0f) {
+            D_i5_801BBEF4[9] = 0.0f;
+        } else {
+            D_i5_801BBEF4[9] = 48.0f;
+            D_i5_801BBEF0[5]--;
+        }
+    }
+    if (D_i5_801BBEF0[6] != 0) {
+        for (i = 0; i < ARRAY_COUNTU(D_i5_801BBF00); i++) {
+            for (j = 0; j < 14; j++) {
+                if (D_i5_801B7770[i][0] == D_i5_801B8160[j]) {
+                    break;
+                }
+            }
+            if (j == 14) {
+                D_i5_801BBF00[i].unk_20 = (Rand_ZeroOne() * 5.0f) - 2.5f;
+            }
+        }
+    }
+
+    for (i = 0; i < 4; i++) {
+        if ((boss->swork[i + 9] > 0) && (boss->swork[i + 9] < 0xB)) {
+            D_i5_801BD668[i + 24] = 1;
+        }
+    }
+
+    if ((boss->state >= 7) && (boss->state < 0xE) && (boss->dmgType == DMG_BEAM) &&
+        (D_i5_801BAA78[D_i5_801B7904[boss->dmgPart]] == 2) && (D_i5_801BBEF0[8] == 0)) {
+        Audio_PlaySfx(0x29121007, boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+        boss->dmgType = DMG_NONE;
+    }
+    if ((boss->state >= 7) && (boss->state < 0xE)) {
+        if (boss->dmgType == DMG_BEAM) {
+            boss->dmgType = DMG_NONE;
+            D_i5_801BD668[D_i5_801B7904[boss->dmgPart]] = D_i5_801B7960[D_i5_801B7904[boss->dmgPart]][1];
+            D_i5_801BD6B0[D_i5_801B7904[boss->dmgPart]] = 0;
+
+            if ((D_i5_801B7904[boss->dmgPart] == 6) && (boss->state == 0xB)) {
+                D_i5_801BD668[D_i5_801B7904[boss->dmgPart]] = 0;
+            }
+
+            if (D_i5_801BAA78[D_i5_801B7904[boss->dmgPart]] != 0) {
+                Audio_PlaySfx(0x2940802C, boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                boss->swork[D_i5_801BAA78[D_i5_801B7904[boss->dmgPart]] + 4] = 0x14;
+                if (boss->swork[D_i5_801BAA78[D_i5_801B7904[boss->dmgPart]] + 8] > 0) {
+                    boss->swork[D_i5_801BAA78[D_i5_801B7904[boss->dmgPart]] + 8] -= boss->damage;
+                    if (boss->swork[D_i5_801BAA78[D_i5_801B7904[boss->dmgPart]] + 8] <= 0) {
+                        boss->swork[D_i5_801BAA78[D_i5_801B7904[boss->dmgPart]] + 8] = 0;
+                        D_i5_801BBEF0[6] = 0x28;
+                        D_i5_801BBEF4[10] = 2.0f;
+                    }
+                }
+                var_v1 = 0;
+                for (i = 0; i < 4; i++) {
+                    if (boss->swork[i + 9] <= 0) {
+                        var_v1++;
+                    }
+                }
+                if ((var_v1 == 3) && (D_i5_801BBEF0[8] == 0)) {
+                    D_i5_801BBF00[39].unk_26 |= 0x11;
+                }
+                if (var_v1 == 4) {
+                    D_i5_801BBEF0[41] = 0;
+                    boss->swork[28] = 1;
+                    boss->swork[0x1B] = 0x50;
```

Comment:
```suggestion
                    boss->swork[27] = 80;
```
other work indices should also be decimal.